model-engine/model_engine_server/domain/use_cases/model_endpoint_use_cases.py

```diff
@@ -215,6 +217,47 @@ def validate_post_inference_hooks(user: User, post_inference_hooks: Optional[Lis
             )
 
 
+def validate_bundle_multinode_compatibility(bundle: ModelBundle, nodes_per_worker: int):
+    """
+    Only some bundles can be multinode compatible.
+    """
+    if nodes_per_worker == 1:
+        return
+    # can type ignore, bundle.flavor is a RunnableImageFlavor/StreamingEnhancedRunnableImageFlavor thus it has worker_command and worker_env
+    if (
+        type(bundle.flavor) in {RunnableImageFlavor, StreamingEnhancedRunnableImageFlavor}
+        and bundle.flavor.worker_command is not None  # type: ignore
+        and bundle.flavor.worker_env is not None  # type: ignore
+    ):
+        return
+    raise ObjectHasInvalidValueException(
+        f"Bundle {bundle.name} is not multinode compatible. It must be a RunnableImage and have _worker_command and _worker_args metadata."
+    )
+
+
+def validate_endpoint_resource_multinode_compatibility(
+    gpu_type: Optional[str],
+    gpus: Optional[int],
+    endpoint_type: ModelEndpointType,
+    nodes_per_worker: int,
+):
+    """
+    Only gpu streaming endpoints can be multinode compatible.
+    """
+    if nodes_per_worker == 1:
+        return
+    if (
+        endpoint_type == ModelEndpointType.STREAMING
+        and gpu_type is not None
+        and gpus is not None
+        and gpus > 0
```

Comment:
is the gpu configuration per worker or total across nodes

Reply:
gpus (and all the rest of the resource requests) are per worker

Reply:
I'm realizing that the term `worker` is slightly overloaded and imprecise since it's defined and used in both `nodes_per_worker` (where it represents the multinode entity) and in `leaderworkerset` (where it represents an individual node).

Could you update our documentation so it's clear what exactly these resources are tied to. 



So

Reply:
Added some docs to the llm engine client side of things